src/Avalonia.Visuals/Rendering/ManagedDeferredRendererLock.cs

```diff
@@ -3,10 +3,11 @@
 
 using System;
 using System.Threading;
+using Avalonia.Utilities;
 
 namespace Avalonia.Rendering
 {
-    public class ManagedDeferredRendererLock : IDeferredRendererLock
+    public class ManagedDeferredRendererLock : DisposableLock, IDeferredRendererLock
```

Comment:
This is kinda weird - you derive from `DisposableLock` but all methods here are just hiding base implementation and it ends up having two different locks.

Reply:
Since DisposableLock class contents are copy-pasted from ManagedDeferredRendererLock, it's most likely an unfinished attempt to extract disposable lock logic to a different class